tests/test_cli_layout_focus_contract.py

```
@@ -0,0 +1,236 @@
+        env=env,
+        timeout=5,
+    )
+    if proc.returncode != 0:
+        merged = f"{proc.stdout}\n{proc.stderr}".strip()
+        raise AssertionError(f"CLI failed ({' '.join(args)}): {merged}")
+    return proc.stdout.strip()
+
+
+def assert_cli_fails(cli: str, socket_path: str, args: list[str], expected: str) -> None:
+    env = dict(os.environ)
+    proc = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)